tests/storage/iso/conftest.py

```diff
@@ -0,0 +1,125 @@
+import pytest
+import logging
+import time
+
+def setup_formatted_and_mounted_disk(host, sr_disk, fs_type, mountpoint):
+    device = '/dev/' + sr_disk
+    logging.info(f">> Format sr_disk {sr_disk} and mount it on host {host}")
+    host.ssh(['mkfs.' + fs_type, "-F", device])
+    host.ssh(['rm', '-rf', mountpoint]) # Remove any existing leftover to ensure rmdir will not fail in teardown
+    host.ssh(['mkdir', '-p', mountpoint])
+    host.ssh(['cp', '-f', '/etc/fstab', '/etc/fstab.orig'])
+    host.ssh(['echo', f'{device} {mountpoint} {fs_type} defaults 0 0', '>>/etc/fstab'])
+    host.ssh(['mount', mountpoint])
```

Comment:
Deux améliorations :
* tenir compte du fs_type pour l'option force de mkfs et raise en cas de FS non supporté par la fonction (on pourra étendre la liste plus tard)
* restaurer fstab si le mount a échoué

De plus, je pense que cette méthode est suffisamment générique pour qu'on la rende accessible à tous les tests, et donc qu'on la mette dans lib/common.py (à défaut de mieux). Même chose pour la fonction de teardown. On pourrait faire ça dans un second commit dédié au refactoring.

Enfin, étant donné qu'une bonne partie du code est dupliqué (en le rendant plus générique) de la fixture `gluster_disk`, il faudrait faire en sorte que la-dite fixture utilise ces deux fonctions. Fixture qu'on pourrait peut-être aussi renommer `formatted_and_mounted_xfs_disk` pour aller avec `formatted_and_mounted_ext4_disk`, et on mettrait ces deux fixtures dans le conftest.py global car elles pourraient bien être utiles dans d'autres tests à l'avenir.

```suggestion
def setup_formatted_and_mounted_disk(host, sr_disk, fs_type, mountpoint):
    if fs_type == 'ext4':
        option_force = '-F'
    elif fs_type == 'xfs':
        option_force = '-f'
    else:
        raise Exception(f"Unsupported fs_type '{fs_type}' in this function")
    assert fs_type in ['ext4', 'xfs']
    device = '/dev/' + sr_disk
    logging.info(f">> Format sr_disk {sr_disk} and mount it on host {host}")
    host.ssh(['mkfs.' + fs_type, option_force, device])
    host.ssh(['rm', '-rf', mountpoint]) # Remove any existing leftover to ensure rmdir will not fail in teardown
    host.ssh(['mkdir', '-p', mountpoint])
    host.ssh(['cp', '-f', '/etc/fstab', '/etc/fstab.orig'])
    host.ssh(['echo', f'{device} {mountpoint} {fs_type} defaults 0 0', '>>/etc/fstab'])
    try:
        host.ssh(['mount', mountpoint])
    except Exception:
        # restore fstab then re-raise
        host.ssh(['cp', '-f', '/etc/fstab.orig', '/etc/fstab'])
        raise
```

Reply:
In addition to this, I think this method is generic enough of to deserve being made available to all tests, in lib/common.py. Same thing for the teardown function. This could be done in a second commit dedicated to refactoring.

Lastly, as most of this code is duplicated from the `gluster_disk` fixture (while at the same time made more generic), that fixture should use the new functions in its implementation.

Reply:
The assert should be enough, why do you have the if before?

Reply:
Actually I wanted to remove the assert and forgot :D. As we already review the supported types with the if, I decided raising in the `else` would be better than repeating the list of supported types.

Reply:
Do i need to modify gluster tests right now ? Or will we do that in another batch of modifications after this PR ?

Reply:
or someone else will modify that ?

Reply:
I think you can do it in this PR, either in the same commit or in a new commit in the same PR.